### PR TITLE
Mes 4503 safety questions component

### PR DESCRIPTION
**Pull request by mgoldson**
## Description
Adds the new Safety Questions component which appears on the Vehicle Checks modal.

- Vehicle Checks validation on WRTC now includes safety questions
- Driving fault badge on WRTC will now include safety question faults

## Checklist

- [x] PR title includes the JIRA ticket number
- [x] Branch is rebased against the latest develop
- [x] Code has been tested manually
- [x] PR link added to JIRA ticket
- [ ] One review from each scrum team
- [ ] Squashed commit contains the JIRA ticket number

## Screenshots (optional)
localhost_8100_(Ipad DES) (2)
localhost_8100_(Ipad DES) (1)
localhost_8100_(Ipad DES)

### CODE REVIEW

File: src/modules/tests/test-data/cat-d/_tests_/test-data.cat-d.selector.spec.ts [hunk 1]
```diff
@@ -0,0 +1,390 @@
...
+  getManoeuvres,
+  hasManoeuvreBeenCompletedCatD,
+  areTellMeQuestionsSelected,
+  areTellMeQuestionsCorrect,
+  hasVehicleChecksBeenCompletedCatD,
+} from '../test-data.cat-d.selector';
+import { Competencies } from '../../test-data.constants';
+import { CompetencyOutcome } from '../../../../../shared/models/competency-outcome';
+import { OutcomeBehaviourMapProvider } from '../../../../../providers/outcome-behaviour-map/outcome-behaviour-map';
+import { behaviourMap } from '../../../../../pages/office/office-behaviour-map';
```
Comment (rossfauldsbjss) [inline src/modules/tests/test-data/cat-d/_tests_/test-data.cat-d.selector.spec.ts]: there is a cat d specific behaviour map. It's pretty much a copy of Cat B+E and Cat C but is the correct one to use for Cat D pages. (look in office page cat d for it's location)

File: src/modules/tests/test-data/cat-d/safety-questions/__tests__/safety-questions.cat-d.selector.spec.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    it('should return true if there are safety checks entered', ()  => {
+      const populatedSafetyQuestions: CatDUniqueTypes.SafetyQuestions =
+        {
+          questions: [{ description: 'test s01', outcome: 'P' }],
+          faultComments: '',
+        };
+      const result = safetyQuestionsExist(populatedSafetyQuestions);
+      expect(result).toBeTruthy();
+    });
+    it('should return false if there are vehicle checks but no outcome selected', ()  => {
```
Comment (rossfauldsbjss) [inline src/modules/tests/test-data/cat-d/safety-questions/__tests__/safety-questions.cat-d.selector.spec.ts]: should be safety checks not vehicle checks.

**Review by rossfauldsbjss — APPROVED** [review]
Great job. Just small change to use cat d behaviour map and that's it.

File: src/modules/tests/test-data/cat-d/_tests_/test-data.cat-d.selector.spec.ts (hunk 1 above)
Comment (mgoldson) [reply src/modules/tests/test-data/cat-d/_tests_/test-data.cat-d.selector.spec.ts]: Looks like the test data selector needs a minor refactor. I have removed some tests which are not applicable to categories other than B and we should pick up the refactor outside of this PR.

File: src/modules/tests/test-data/cat-d/safety-questions/__tests__/safety-questions.cat-d.selector.spec.ts (hunk 2 above)
Comment (mgoldson) [reply src/modules/tests/test-data/cat-d/safety-questions/__tests__/safety-questions.cat-d.selector.spec.ts]: Good spot - thanks!

**Review by faggc — APPROVED** [review]

**Pull request merged**